perllib/FixMyStreet/Roles/Bottomline.pm

```diff
@@ -91,4 +91,33 @@ sub waste_payment_type {
     return ($category, $sub_type);
 }
 
+sub _process_reference {
+    my ($self, $payer) = @_;
+
+    my ($id, $uprn) = $payer =~ /^@{[$self->waste_payment_ref_council_code()]}-(\d+)-(\d+)/;
+
+    return (undef, undef) unless $id;
+    my $origin = FixMyStreet::DB->resultset('Problem')->find($id);
+
+    if ( !$origin ) {
+        $self->log("no matching origin sub for id $id");
+        return (undef, undef);
+    }
+
+    $uprn = $origin->get_extra_field_value('uprn');
```

Comment:
I'm not sure what this section is for - presumably if we've already got the report (from its ID) by here, we don't need to check the UPRN as well?

Reply:
It's more that we use the UPRN later on in processing when creating any new thing. Although I guess we don't really need to as we can just copy it from the existing one.